src/index.ts

```diff
@@ -0,0 +1,52 @@
+import { resolve, join } from 'path';
+import { readJson } from 'fs-extra';
+import * as core from '@actions/core';
+import * as github from '@actions/github';
+import remoteTags from 'remote-git-tags';
+
+const REQUIRED_TAG_FORMAT = '$version';
+
+// eslint-disable-next-line camelcase
+const cloneUrl: string = github.context.payload.repository?.clone_url;
+const tagFormat: string = core.getInput('tag-format');
+
+async function getVersion(): Promise<string> {
```

Comment:
Do I need to also specify that this could return an error?

Reply:
No, throwing an error is assumed functionality of a function